platform/eureka/client_test.go

```diff
@@ -0,0 +1,82 @@
+package eureka
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func readFile(t *testing.T, filename string) []byte {
+	data, err := ioutil.ReadFile(filename)
+	if err != nil {
+		t.Error(err)
+	}
+	return data
+}
+
+func TestClient(t *testing.T) {
+	clientTests := []struct {
+		context    string
+		data       []byte
+		statusCode int
+		apps       []*application
+		shouldErr  bool
+	}{
+		{
+			context:    "no applications",
+			data:       readFile(t, "testdata/eureka-no-apps.json"),
+			statusCode: http.StatusOK,
+			apps:       make([]*application, 0),
+		},
+		{
+			context: "multiple applications",
+			data:    readFile(t, "testdata/eureka-apps.json"),
+			apps: []*application{
+				{
+					Name: appName("foo.bar.local"),
```

Comment:
Nit: This is fine for testing, but I don't think it is common to place a full DNS entry in the Eureka application name.